Author: Ben Hutchings <benh@debian.org>
Description: color: Handle NO_COLOR environment variable in default_color_opt()
Origin: commit:f0076a016cf7926f39c47b95dae0002249c082dc
Forwarded: https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/commit/?id=f0076a016cf7926f39c47b95dae0002249c082dc
--- a/lib/color.c
+++ b/lib/color.c
@@ -83,6 +83,13 @@
 
 int default_color_opt(void)
 {
+	const char *no_color;
+
+	/* If NO_COLOR has a non-empty value, coloured output is never wanted */
+	no_color = getenv("NO_COLOR");
+	if (no_color && *no_color)
+		return COLOR_OPT_NEVER;
+
 	return CONF_COLOR;
 }
 
